sdk/webpubsub/azure-messaging-webpubsubservice/tests/test_reverse_proxy.py

```diff
@@ -9,31 +9,45 @@
-    credential = AzureKeyCredential("abcdabcdabcdabcdabcdabcdabcdabcd")
-    client = WebPubSubServiceClient(wps_endpoint, "Hub", credential, reverse_proxy_endpoint=apim_endpoint)
-    request = build_send_to_all_request('Hub', content='test_webpubsub_send_request', content_type='text/plain')
+class WebpubsubReverseProxyTest(WebpubsubTest):
 
-    with pytest.raises(ValueError) as ex:
-        client.send_request(request, raw_request_hook=_callback)
-    assert "Success!" in str(ex.value)
+    def test_reverse_proxy_endpoint_redirection_azure_key_credential(self):
+        def _callback(pipeline_request):
+            assert pipeline_request.http_request.url.startswith("https://apim.contoso.com/")
+            raise ValueError("Success!")
+        wps_endpoint = "https://wps.contoso.com/"
+        apim_endpoint = "https://apim.contoso.com/"
+        credential = AzureKeyCredential("abcdabcdabcdabcdabcdabcdabcdabcd")
+        client = WebPubSubServiceClient(wps_endpoint, "Hub", credential, reverse_proxy_endpoint=apim_endpoint)
+        request = build_send_to_all_request('Hub', content='test_webpubsub_send_request', content_type='text/plain')
 
-def test_reverse_proxy_endpoint_redirection_identity():
-    def _callback(pipeline_request):
-        assert pipeline_request.http_request.url.startswith("https://apim.contoso.com/")
-        raise ValueError("Success!")
-    wps_endpoint = "https://wps.contoso.com/"
-    apim_endpoint = "https://apim.contoso.com/"
-    credential = FakeTokenCredential()
-    client = WebPubSubServiceClient(wps_endpoint, "Hub", credential, reverse_proxy_endpoint=apim_endpoint)
-    request = build_send_to_all_request('Hub', content='test_webpubsub_send_request', content_type='text/plain')
+        with pytest.raises(ValueError) as ex:
+            client.send_request(request, raw_request_hook=_callback)
+        assert "Success!" in str(ex.value)
 
-    with pytest.raises(ValueError) as ex:
-        client.send_request(request, raw_request_hook=_callback)
-    assert "Success!" in str(ex.value)
+    def test_reverse_proxy_endpoint_redirection_identity(self):
+        def _callback(pipeline_request):
+            assert pipeline_request.http_request.url.startswith("https://apim.contoso.com/")
+            raise ValueError("Success!")
+        wps_endpoint = "https://wps.contoso.com/"
+        apim_endpoint = "https://apim.contoso.com/"
+        credential = FakeTokenCredential()
+        client = WebPubSubServiceClient(wps_endpoint, "Hub", credential, reverse_proxy_endpoint=apim_endpoint)
+        request = build_send_to_all_request('Hub', content='test_webpubsub_send_request', content_type='text/plain')
+
+        with pytest.raises(ValueError) as ex:
+            client.send_request(request, raw_request_hook=_callback)
+        assert "Success!" in str(ex.value)
+
+    @WebpubsubPowerShellPreparer()
+    def test_reverse_proxy_call(self, webpubsub_connection_string, webpubsub_reverse_proxy_endpoint):
```

Comment:
ignore the whole diff in this file except the addition of this test.

(the above diff is because I added the existing tests to a testcase class